_src/_posts/2017-01-31-learning-rtb.markdown

```diff
@@ -0,0 +1,205 @@
+---
+layout: post
+title:  Learning from the RTB market
+date:   2017-01-31
+tag: data-science
+categories:
+  - data-science
+keywords: "python, RTB, machine-learning"
+author: memeplex
+---
+
+As you might already know, [Jampp][jampp] is a demand-side platform ([DSP][dsp])
+that allows companies to promote their mobile applications by leveraging
+real-time bidding ([RTB][rtb]) technologies in order to buy digital advertising
+from multiple inventory sources and exchanges. During an RTB transaction, an
+auction is announced and any interested bidder has to answer with a bid price
+within a time constraint of about 100 milliseconds. The bidder that wins the
+auction pays the second-highest price and obtains the right to print a creative
+(which is just jargon for displaying an ad) on a publisher site. If the banner
+is clicked by the user further goal events might be tracked, v.g. the
+application may be installed or opened. Obtaining a goal event is called a
+conversion.
+
+One main component of our platform is the machine learning subsystem that feeds
+the bidding subsystem with predictive models of conversion rates for different
+events of interest (clicks, installs, opens, etc.). A conversion rate for an
+event is the conditional probability of getting that event given the set of
+features that characterizes the current RTB transaction and given that we
+already printed a creative targeting that event. Why is it so important for our
+business to have good estimates of these rates? Simply because we sell
+conversions but we pay for impressions. For example, suppose one client pays us
+$\$1$ per install and we estimate the install conversion rate to be around $0.1$
+(just cheating... real world figures are well under a dismaying $0.001$ `:(`).
+Then, assuming our creative is going to get printed, we should expect an average
+profit of $\$0.1 - c$, where $c$ is the expected cost of the impression. If our
+rate estimates are grossly miscalculated we will be losing money by valuating
+the expected income too low (thus missing opportunities with $c < \$0.1$) or too
```

Comment:
Isn't it the opposite? "0.1" is hardcoded in the example, but it's actually the value calculated by us, which can be wrong. In the "missing opportunities" entry, If predicted_rate == 0.1, and $c < predicted_rate (say c == 0.05) => we don't bid. But we should have bid, if we had calculated  predicted_rate as 0.05 (the "real" value). Then, the "expected conversion rate" and (proportionally) the "expected income" are too high (it is the double). Same happens with the "too high" section, it should be "too low".